### PR TITLE
[WIP] lossless-cut: Init at version 3.23.7

### LINKED ISSUE #91685: Packaging requests: `lossless-cut`
**Project description**
_LosslessCut the swiss army knife of lossless video/audio editing that makes use of FFmpeg_
A ready-to-merge binary build: #108512 
The draft PR for the source-build: #91683

**Metadata**

* homepage URL: https://mifi.no/losslesscut
* source URL: https://github.com/mifi/lossless-cut
* license: mit
* platforms: unix

### PR DESCRIPTION
**Pull request by ShamrockLee**
This is a draft to package [lossless-cut](https://github.com/mifi/lossless-cut) .
As a beginner, I may need help from others often.
Any advise, contributions or corrections are more than welcome. If you would like to be added as a maintainer, please go ahead or inform me.

###### Motivation for this change
LosslessCut is an intuitive GUI application that cuts videos/audios losslessly.
If applied, we will be able to natively install this app.

###### Things done
Create a simple `default.nix` that makes use of `mkYarnPackage`.
Create `yarn.nix` using `yarn2nix`.
Add entry lossless-cut to `all-packages.nix`.

Takes 50 minutes to build on my slow usb-stick-based NixOS system,
and most of the time is spent on the node dependencies.
Closure size: 1,614,691,296 bytes, where
$out/libexec/lossless-cut/node_modules occupies 686,152,094 bytes.

Problem: $out/bin is empty.
TODO: Add something to the `installPhase`.

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/video/lossless-cut/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  yarnNix = ./yarn.nix;
+  yarnLock = "${src}/yarn.lock";
+  packageJSON = "${src}/package.json";
+
+  meta = with stdenv.lib; {
+    description = "The swiss army knife of lossless video/audio editing";
+    license = licenses.mit;
+    homepage = "https://mifi.no/losslesscut";
+    maintainers = with maintainers; [ ShamrockLee ];
```
Comment (veprbl) [inline pkgs/applications/video/lossless-cut/default.nix:22]: You need to add yourself to maintainers/maintainer-list.nix, preferably do that in a separate commit.

Comment (ShamrockLee) [reply pkgs/applications/video/lossless-cut/default.nix:22]: The problem is that I added myself to the `maintainer-list.nix` in another new package commit waiting to be merged (i.e. #91270).
I don't know if it would cause conflicts.

Comment (ShamrockLee) [reply pkgs/applications/video/lossless-cut/default.nix:22]: I just added myself to the `maintainer-list.nix` to make it evaluatable.

File: pkgs/applications/video/lossless-cut/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  src = fetchFromGitHub {
+    owner = "mifi";
+    repo = name;
+    rev = "v${version}";
+    sha256 = "14vfmidj0m11vd1asghmcxwj102c468n6wb4swagnjzrxqh22xa3";
+  };
+
+  yarnNix = ./yarn.nix;
+  yarnLock = "${src}/yarn.lock";
+  packageJSON = "${src}/package.json";
```
Comment (veprbl) [inline pkgs/applications/video/lossless-cut/default.nix:16]: I guess, this relies on Import From Derivation, which is not allowed in nixpkgs. Seems like other packages put this file into nixpkgs directly.

Also these two lines seem to be set by default:
https://github.com/NixOS/nixpkgs/blob/db5bbef31fa05b9634fa6ea9a5afbea463da88ea/pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix#L226-L227

Comment (ShamrockLee) [reply pkgs/applications/video/lossless-cut/default.nix]: I removed the three lines specifying `yarnNix`, `yarnLock` and `packageJSON` and deleted `yarn.nix`.

Comment (veprbl) [reply pkgs/applications/video/lossless-cut/default.nix]: I didn't mean to say that the default settings will work for nixpkgs. You might need to copy file to nixpkgs repo at least for `packageJSON` to avoid IFD.

**Comment by ShamrockLee** [comment]
Close due to difficulties faced during source-packaging Electron apps.
If you are interested in making LosslessCut available in Nixpkgs, please help review/merge th binary build PR: #108512 :grinning:
It would still be great if we can build Electron apps from source with Nix one day.
